internal/resources/syntheticmonitoring/resource_check.go

```diff
@@ -119,6 +119,23 @@ var (
 				MaxItems:    1,
 				Elem:        syntheticMonitoringCheckSettingsMultiHTTP,
 			},
+			"scripted": {
+				Description: "Settings for scripted check. See https://grafana.com/docs/grafana-cloud/testing/synthetic-monitoring/create-checks/checks/k6/.",
+				Type:        schema.TypeSet,
+				Optional:    true,
+				MaxItems:    1,
+				Elem:        syntheticMonitoringCheckSettingsScripted,
+			},
+		},
+	}
```

Comment:
@julienduchesne, Thanks for the quick response but it would be nice if we can take in a file. For example,

instead of 

```
```terraform
data "grafana_synthetic_monitoring_probes" "main" {}
resource "grafana_synthetic_monitoring_check" "scripted" {
  job     = "Scripted defaults"
  target  = "scripted target"
  enabled = false
  probes = [
    data.grafana_synthetic_monitoring_probes.main.probes.Paris,
  ]
  labels = {
    foo = "bar"
  }
  settings {
    scripted {
      script = "console.log('Hello, world!')"
    }
  }
}
```
something like this would be nicer 

```
```terraform
data "grafana_synthetic_monitoring_probes" "main" {}
resource "grafana_synthetic_monitoring_check" "scripted" {
  job     = "Scripted defaults"
  target  = "scripted target"
  enabled = false
  probes = [
    data.grafana_synthetic_monitoring_probes.main.probes.Paris,
  ]
  labels = {
    foo = "bar"
  }
  settings {
    scripted {
      script = "/path/to/js-script.js"
    }
  }
}
```
Happy to look into this and contribute a Pr if needed.

Reply:
You can use the file function for that: https://developer.hashicorp.com/terraform/language/functions/file
File paths don't work well in Terraform because they don't take into consideration the content of the file

Reply:
True, totally Forgot about that.

Reply:
I updated the example to use the `local_file` data source. This reacts correctly to changes.

Reply:
That's not what local_file is for. `local_file` generates a file. The syntax is just `file()`

Reply:
Nooo, not local_file. It's just `file(...)`. `local_file` generates a file which is not what we want here. I've fixed it here: https://github.com/grafana/terraform-provider-grafana/pull/1632
And I changed the repo rules so that failing CI prevents merge